finalfrontier/src/vocab.rs

```diff
@@ -213,10 +213,29 @@ impl From<SubwordVocab> for VocabWrap {
         .into()
     }
 }
+/// Index representation.
+///
+/// This enum represents possible types of indices, e.g. in a vocabulary with subwords, the `Multi`
+/// variant includes all indices associated with a key.
+#[derive(Clone)]
+pub enum Idx {
```

Comment:
This approach has the downside that without subwords, indices are still ptr + cap + len size due to the largest enum variant (+ possible padding / alignment).